connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java

```diff
@@ -2622,4 +2621,135 @@ void testRetrieveByProduct() {
         assertEquals(pageNumber, capturedPage.getPageNumber());
         verifyNoMoreInteractions(institutionRepository);
     }
+
+    @Test
+    void findBrokers() {
+        final String productId = "productId";
+
+        BillingEntity billingEntity = new BillingEntity();
+        billingEntity.setPublicServices(true);
+        billingEntity.setRecipientCode("Recipient Code");
+        billingEntity.setVatNumber("42");
+
+        DataProtectionOfficerEntity dataProtectionOfficerEntity = new DataProtectionOfficerEntity();
+        dataProtectionOfficerEntity.setAddress("42 Main St");
+        dataProtectionOfficerEntity.setEmail("jane.doe@example.org");
+        dataProtectionOfficerEntity.setPec("Pec");
+
+        PaymentServiceProviderEntity paymentServiceProviderEntity = new PaymentServiceProviderEntity();
+        paymentServiceProviderEntity.setAbiCode("Abi Code");
+        paymentServiceProviderEntity.setBusinessRegisterNumber("42");
+        paymentServiceProviderEntity.setLegalRegisterName("Legal Register Name");
+        paymentServiceProviderEntity.setLegalRegisterNumber("42");
+        paymentServiceProviderEntity.setVatNumberGroup(true);
+
+        OnboardingEntity onboardingEntity = new OnboardingEntity();
+        onboardingEntity.setBilling(billingEntity);
+        onboardingEntity.setClosedAt(null);
+        onboardingEntity.setContract("Contract");
+        onboardingEntity.setCreatedAt(null);
+        onboardingEntity.setPricingPlan("Pricing Plan");
+        onboardingEntity.setProductId(productId);
+        onboardingEntity.setStatus(RelationshipState.PENDING);
+        onboardingEntity.setTokenId("42");
+        onboardingEntity.setUpdatedAt(null);
+
+        InstitutionEntity institutionEntity = new InstitutionEntity();
+        institutionEntity.setAddress("42 Main St");
+        institutionEntity.setAttributes(new ArrayList<>());
+        institutionEntity.setBilling(billingEntity);
+        institutionEntity.setBusinessRegisterPlace("Business Register Place");
+        institutionEntity.setCreatedAt(null);
+        institutionEntity.setDataProtectionOfficer(dataProtectionOfficerEntity);
+        institutionEntity.setDescription("The characteristics of someone or something");
+        institutionEntity.setDigitalAddress("42 Main St");
+        institutionEntity.setExternalId("42");
+        institutionEntity.setGeographicTaxonomies(new ArrayList<>());
```

Comment:
we could avoid to replicate the same definition of an object for each test method....you might implement an utility test (es. createDummy PaymentServiceProvider) that create a dummy object with default value for each attribute and using that when you need a dummy object for test.....look at https://github.com/pagopa/selfcare-ms-core/blob/main/web/src/test/java/it/pagopa/selfcare/mscore/web/TestUtils.java